hr_expense_cancel/model/hr_expense_cancel.py

```diff
@@ -0,0 +1,20 @@
+# coding: utf-8
+
+from openerp import api, models
+
+
+class HrExpenseExpense(models.Model):
+    _inherit = "hr.expense.expense"
+
+    @api.multi
+    def expense_canceled(self):
+        account_move_line_obj = self.env['account.move.line']
+        res = super(HrExpenseExpense,
+                    self).expense_canceled()
+        for expense in self:
+            if expense.account_move_id:
+                all_move_line_ids = [x.id for x in
+                                     expense.account_move_id.line_id]
+                account_move_line_obj._remove_move_reconcile(all_move_line_ids)
```

Comment:
You shouldn't remove reconcile lines without any advertising. I think this module should only try to remove the lines, and if there is any line reconciled, raises an error, so you manually remove that reconcilement.